Userland/Libraries/LibJS/Runtime/ArrayBuffer.h

```diff
@@ -50,23 +56,25 @@ class ArrayBuffer : public Object {
     ByteBuffer const& buffer() const { return buffer_impl(); }
 
     // Used by allocate_array_buffer() to attach the data block after construction
-    void set_buffer(ByteBuffer buffer) { m_buffer = move(buffer); }
+    void set_data_block(DataBlock block) { m_data_block = move(block); }
 
     Value detach_key() const { return m_detach_key; }
     void set_detach_key(Value detach_key) { m_detach_key = detach_key; }
 
-    void detach_buffer() { m_buffer = Empty {}; }
+    void detach_buffer() { m_data_block.buffer = Empty {}; }
 
     // 25.1.2.2 IsDetachedBuffer ( arrayBuffer ), https://tc39.es/ecma262/#sec-isdetachedbuffer
     bool is_detached() const
     {
         // 1. If arrayBuffer.[[ArrayBufferData]] is null, return true.
-        if (m_buffer.has<Empty>())
+        if (m_data_block.buffer.has<Empty>())
             return true;
         // 2. Return false.
         return false;
     }
 
+    bool is_shared_array_buffer() const { return m_data_block.is_shared; }
```

Comment:
This is not what the spec says for https://tc39.es/ecma262/#sec-issharedarraybuffer

Reply:
Indeed, implemented it according to spec.